### PR TITLE
[genomic_browser] upgraded module for react & using DataFramework

### PR DESCRIPTION
**Pull request by maltheism**
## Brief summary of changes

Upgraded the genomic_browser module to use FilterDataTable components and the DataFramework on the backend. A lot of the backend code isn't modified and using the same code from before. I did move the uploading file code to a new class and restructure the class files for replicating the directory structure of the conflict resolver DataFramework PR.

Note: The genomic_browser module is slow from observing all the data that's retrieved from the database. I don't believe this PR should fix that problem.

#### Testing instructions (if applicable)

1. checkout pr.
2. test with rasinbread data or user data.

### CODE REVIEW

**Comment by maltheism** [comment]
See #5261 previous PR.

**Comment by johnsaigle** [comment]
Why did you close the other pull request?

**Comment by maltheism** [comment]
@johnsaigle didn't feel like rebasing it.

**Comment by johnsaigle** [comment]
Creating new PRs can make things difficult because the review discussion gets hidden and the commit history gets lost.

**Comment by maltheism** [comment]
@johnsaigle I understand but didn't want to risk messing up the rebase and since there are lots of files in this PR.

File: modules/genomic_browser/php/endpoints/filemanager.class.inc [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+     * Provide an array of distinct value from a column in
+     * the specified table. The first element has a null key and
+     * its value is 'Any'.
+     *
+     * @param string $table  The tables name
+     * @param string $column The column name
+     *
+     * @return array An array where value are keys as well as values
+     */
+    function getDistinctValues( $table, $column)
```
Comment (johnsaigle) [inline modules/genomic_browser/php/endpoints/filemanager.class.inc]:
```suggestion
    function getDistinctValues(string $table, string $column)
```

File: modules/genomic_browser/php/endpoints/filemanager.class.inc [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+     * @param array                  $values  The form values.
+     *
+     * @return array
+     */
+    private function _verifyPermissions(
+        ServerRequestInterface $request,
+        array $values
+    ) : array {
+        $user = $request->getAttribute('user');
+        if (!$user->hasAnyPermission(
```
Comment (johnsaigle) [inline modules/genomic_browser/php/endpoints/filemanager.class.inc]:
```suggestion
        if (!$this->_hasAccess($user)) {
```

File: modules/genomic_browser/php/endpoints/filemanager.class.inc [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+            [
+             'genomic_browser_view_site',
+             'genomic_browser_view_allsites',
+            ]
+        )
```
Comment (johnsaigle) [inline modules/genomic_browser/php/endpoints/filemanager.class.inc]: These lines can be deleted when the above suggestion is applied.

File: modules/genomic_browser/php/endpoints/gwasbrowser.class.inc [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+     * @param array                  $values  The form values.
+     *
+     * @return array
+     */
+    private function _verifyPermissions(
+        ServerRequestInterface $request,
+        array $values
+    ) : array {
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('genomic_browser_view_site')
```
Comment (johnsaigle) [inline modules/genomic_browser/php/endpoints/gwasbrowser.class.inc]: Replace with `hasAccess()` check, as above

File: modules/genomic_browser/php/endpoints/methylation.class.inc [hunk 5]
```diff
@@ -0,0 +1,516 @@
...
+     * @param array                  $values  The form values.
+     *
+     * @return array
+     */
+    private function _verifyPermissions(
+        ServerRequestInterface $request,
+        array $values
+    ) : array {
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('genomic_browser_view_site')
```
Comment (johnsaigle) [inline modules/genomic_browser/php/endpoints/methylation.class.inc]: Same here

File: modules/genomic_browser/php/endpoints/profiles.class.inc [hunk 6]
```diff
@@ -0,0 +1,279 @@
...
+     * @param array                  $values  The form values.
+     *
+     * @return array
+     */
+    private function _verifyPermissions(
+        ServerRequestInterface $request,
+        array $values
+    ) : array {
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('genomic_browser_view_site')
```
Comment (johnsaigle) [inline modules/genomic_browser/php/endpoints/profiles.class.inc]: Same here.

File: modules/genomic_browser/php/endpoints/snpbrowser.class.inc [hunk 7]
```diff
@@ -0,0 +1,474 @@
...
+     * @param array                  $values  The form values.
+     *
+     * @return array
+     */
+    private function _verifyPermissions(
+        ServerRequestInterface $request,
+        array $values
+    ) : array {
+        $user = $request->getAttribute('user');
+        if (!$user->hasPermission('genomic_browser_view_site')
```
Comment (johnsaigle) [inline modules/genomic_browser/php/endpoints/snpbrowser.class.inc]: same here. I'm not going to keep marking them but please apply the same change in all files.

**Comment by maltheism** [comment]
Hey @johnsaigle, I think I've resolved the permission code in the latest commit. Please let me know if I missed anything. Thanks for the help.

File: modules/genomic_browser/php/endpoints/snpbrowser.class.inc [hunk 8]
```diff
@@ -447,8 +424,7 @@ class SnpBrowser extends Endpoint implements ETagCalculator
      */
     function _hasAccess(\User $user) : bool
     {
-        return ($user->hasPermission('genomic_browser_view_allsites')
-            || $user->hasPermission('genomic_browser_view_site'));
+        return ($this->_hasAccess($user));
```
Comment (johnsaigle) [inline modules/genomic_browser/php/endpoints/snpbrowser.class.inc]: Revert this change please.

Comment (maltheism) [reply modules/genomic_browser/php/endpoints/snpbrowser.class.inc]: @johnsaigle fixed

**Comment by johnsaigle** [comment]
Looks like the integration tests need to be updated.

**Comment by maltheism** [comment]
Closing PR in favour of https://github.com/aces/Loris/pull/5835. I had trouble rebasing.
